builder/relinker/relinker.py

```
@@ -0,0 +1,387 @@
+        self.iram1_exclude = '    *(EXCLUDE_FILE(%s %s) .iram1.*) *(EXCLUDE_FILE(%s %s) .iram1)' % \
+                             (self.filter.add(), ' '.join(iram1_exclude), \
+                              self.filter.add(), ' '.join(iram1_exclude))
+        self.iram1_include = '\n'.join(iram1_include)
+        self.flash_include = '\n'.join(flash_include)
+
+        logging.debug('IRAM1 Exclude: %s'%(self.iram1_exclude))
+        logging.debug('IRAM1 Include: %s'%(self.iram1_include))
+        logging.debug('Flash Include: %s'%(self.flash_include))
+
+    def __replace__(self, lines):
+        def is_iram_desc(l):
```

Ruff (0.15.7)

[error] Ambiguous variable name: `l`

(E741)